packages/formik/src/Formik.tsx

```diff
@@ -939,10 +972,43 @@ export function useFormik<Values extends FormikValues = FormikValues>({
           field.value = field.value || [];
           field.multiple = true;
         }
+
+        if (type !== 'radio' && type !== 'checkbox' && !!format) {
+          if (formatOnBlur === true) {
+            if (touchedState === true) {
+              field.value = format(field.value);
+            }
+          } else {
+            field.value = format(field.value);
+          }
+        }
+
+        // We incorporate the fact that we know the `name` prop by scoping `onChange`.
+        // In addition, to support `parse` fn, we can't just re-use the OG `handleChange`, but
+        // instead re-implement it's guts.
+        if (type !== 'radio' && type !== 'checkbox') {
+          field.onChange = (eventOrValue: React.ChangeEvent<any> | any) => {
```

Comment:
Problem is that checkbox relies on current state for getValuesForCheckbox(). Technically we would need to provide that. What do you think?

Reply:
If that's the case, I think we can provide `field.defaultParseFn` or something.

Reply:
Oh, you mean in the event the user wants to `getValuesForCheckbox` in userland? hmm.. maybe we'll need to pass formikBag as a third argument, or expose the field's complete value in some way.

If you can figure out a way to use `parse` to update a multi checkbox field, I think that's something that would be required of the parse function. I definitely use that in my onParse function for a custom CheckboxGroup that I use.

Reply:
I've revised my thoughts here, and I agree that onChange should handle the array parsing for checkboxes and multi-selects. But getValueOrEvent should return an unparsed single value, with the default parse function for a number or range field running parseFloat for backwards compatibility.

Reply:
FWIW currently, in the app, I'm working on I have to both serialize and parse numbers to make sure they're expected values when Formik is transforming them correctly and GraphQL knows its an Int.

I get `initialValues` from query parameters in the URL, which are all strings. Then I need to parse them into Integers if they're set, and if they're not, I need to cast them to `''` and then, I need to make sure they're numbers because they might not be to then tell GraphQL "this is an int", this would have been a lot easier had Formik not changed the values to a number. because then I could just cast them when making the query to GraphQL.

Reply:
@lifeiscontent you can override this functionality yourself with `parse={value => value}`

we need to keep the default parsing ints for backwards compatibility for now.